R/get_schema.R

```diff
@@ -60,7 +60,16 @@ get_schema.Id <- function(obj,  ...) {
 #' @rdname get_schema
 get_schema.PqConnection <- function(obj, temporary = FALSE,  ...) {
   if (isTRUE(temporary))  {
-    return(DBI::dbGetQuery(obj, "SELECT nspname FROM pg_namespace WHERE oid = pg_my_temp_schema();")$nspname)
+    temp_schema <- DBI::dbGetQuery(obj, "SELECT nspname FROM pg_namespace WHERE oid = pg_my_temp_schema();")$nspname
+
+    # If no temporary tables have been created, the temp schema has not been determined yet
+    if (identical(temp_schema, character(0))) {
+      temp_table <- dplyr::copy_to(dest = obj, df = mtcars, name = unique_table_name(), temporary = TRUE)
+      defer_db_cleanup(temp_table)
+      temp_schema <- DBI::dbGetQuery(obj, "SELECT nspname FROM pg_namespace WHERE oid = pg_my_temp_schema();")$nspname
+    }
+
+    return(temp_schema)
```

Comment:
From the [Postgres 16 documentation](https://www.postgresql.org/docs/16/runtime-config-client.html#GUC-SEARCH-PATH):

> [The current session's temporary-table schema] can be explicitly listed in the path by using the alias `pg_temp`.

```suggestion
    return("pg_temp")
```

Reply:
Sure, lets try to see if it works. Would be a lot simpler :)